app/views/shared/components/_course_lessons.html.erb

```diff
@@ -57,25 +54,34 @@
             <% end %>
             <span><%= lesson.title %></span>
           </div>
-          <%= link_to course_module_lesson_path(course, course_module, lesson), class: "hover:cursor-pointer" do %>
-            <span class="icon icon-play-lite h-5 w-5 "></span>
-          <% end %>
+
+          <div class="flex gap-6 items-center">
+            <% if policy(@course).edit? %>
+              <%= link_to edit_course_module_lesson_path(course, course_module, lesson), class: "link-text" do %>
+                <%= render "shared/components/button_default_small", label: "Edit", icon_name: "icon-edit" %>
+              <% end %>
+            <% end %>
+            <%= link_to course_module_lesson_path(course, course_module, lesson), class: "hover:cursor-pointer" do %>
+              <span class="icon icon-play-lite h-5 w-5 "></span>
+            <% end %>
+          </div>
+        
         </li>
       <% end %>
-  <!-- TODO: Item with progress bar example -->
-  <!--    <li class="flex items-center justify-between border-t px-6 py-4 text-sm">-->
-  <!--      <div class="flex items-center gap-4">-->
-  <!--        <span class="icon bg-gold border-slate-grey icon-incomplete h-5 w-5 items-center border"></span>-->
-  <!--        <span>Item with progress bar example</span>-->
-  <!--        <div class="flex items-center gap-4 pl-8">-->
-  <!--          <div class="bg-primary2 h-1 w-[252px] rounded-full">-->
-  <!--            <div class="bg-gray5 h-1 rounded-full" style="width: 16%"></div>-->
-  <!--          </div>-->
-  <!--          <span class="text-gray5 text-sm font-semibold">16%</span>-->
-  <!--        </div>-->
-  <!--      </div>-->
-  <!--      <span class="icon icon-play h-5 w-5"></span>-->
-  <!--    </li>-->
+          <!-- TODO: Item with progress bar example -->
```

Comment:
Is this automatic on save or you did manually indent ?

Reply:
this happens on save I think ,and not able to revert back how it was before